ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/PullQueryIQv2FunctionalTest.java

```diff
@@ -359,6 +364,146 @@ public void shouldDoRangeQueriesUpperBound() {
         ImmutableList.of(key3, 1), ImmutableList.of(key4, 1)));
   }
 
+  @Test
```

Comment:
I think we should add a couple of cases for the Session window as well to be complete